cbmc/proofs/mqtt_state.c

```diff
@@ -285,41 +285,64 @@ bool valid_IotMqttNetworkInfo( const IotMqttNetworkInfo_t *pInfo )
  * IotMqttConnectInfo
  ****************************************************************/
 
+void bound_IotMqttConnectInfo( IotMqttConnectInfo_t *pInfo )
+{
+#ifdef SUBSCRIPTION_COUNT_MAX
+  __CPROVER_assume( pInfo->previousSubscriptionCount < SUBSCRIPTION_COUNT_MAX );
```

Comment:
Not at all clear we want to do this for all proofs.  This is a variable all proofs define.

Reply:
`IotMqttConnectInfo_t` contains `pPreviousSubscriptions`, which is a `IotMqttSubscription_t`; therefore, before we nondeterministically allocate the previous subscription list, we must define a bound for `previousSubscriptionCount`. The previous code did that on the validity function.

Reply:
OK